libs/openFrameworks/types/ofBaseTypes.h

```diff
@@ -1786,67 +1786,371 @@ class ofBaseRenderer{
+	/// \brief Draw a texture with this renderer.
+	/// \param image The texture to draw with this renderer.
+	/// \param x The x coordinate to use when drawing the texture with this
+	/// renderer.
+	/// \param y The y coordinate to use to draw the texture with this renderer.
+	/// \param z The z coordinate to use to draw the texture with this renderer.
+	/// \param w The width to use to draw the the texture with this renderer.
+	/// \param h The height to use to draw the the texture with this renderer.
+	/// \param sx The subsection x axis offset within the texture.
+	/// \param sy The subsection y axis offset within the texture.
+	/// \param sw The subsection width offset within the texture.
+	/// \param sh The subsection height offset within the texture.
 	virtual void draw(const ofTexture & image, float x, float y, float z, float w, float h, float sx, float sy, float sw, float sh) const=0;
+	/// \brief Draw vertices from a vertext buffer with this renderer.
+	///
+	///	\p drawMode may be any of the OpenGL primitive draw modes:
+	///		GL_POINTS,
+	///		GL_LINE_STRIP,
+	///		GL_LINE_LOOP,
+	///		GL_LINES,
+	///		GL_TRIANGLE_STRIP,
+	///		GL_TRIANGLE_FAN,
+	///		GL_TRIANGLES
+	///
+	/// ofGetGLPrimitiveMode() can also be used to get the GLuint draw mode
+	/// from an ofPrimitiveMode.
+	///
+	/// \param vbo The vertext buffer object to draw vertices from with this
+	/// renderer.
+	/// \param drawMode OpenGL primitive draw mode to use when drawing the vbo's
+	/// vertices with this renderer.
+	/// \param first The index of the first vertex to draw from \p vbo.
+	/// \param total The total number of indices to draw from \p vbo.
+	/// \sa ofGetGLPrimitiveMode()
+	/// \sa glDrawArrays()
+	/// \sa https://www.khronos.org/opengles/sdk/docs/man/xhtml/glDrawArrays.xml
 	virtual void draw(const ofVbo & vbo, GLuint drawMode, int first, int total) const=0;
+	/// \brief Draw vertices from a vertext buffer with this renderer.
+	///
+	///	\p drawMode may be any of the OpenGL primitive draw modes:
+	///		GL_POINTS,
+	///		GL_LINE_STRIP,
+	///		GL_LINE_LOOP,
+	///		GL_LINES,
+	///		GL_TRIANGLE_STRIP,
+	///		GL_TRIANGLE_FAN,
+	///		GL_TRIANGLES
+	///
+	/// ofGetGLPrimitiveMode() can also be used to get the GLuint draw mode
+	/// from an ofPrimitiveMode.
+	///
+	/// \param vbo The vertext buffer object to draw vertices from with this
+	/// renderer.
+	/// \param drawMode OpenGL primitive draw mode to use when drawing the vbo's
+	/// vertices with this renderer.
+	/// \param amt The number of elements to be rendered.
+	/// \param offsetelements The number of elements (indices) offset from the
+	/// first element to start drawing from.
+	/// \sa ofGetGLPrimitiveMode()
+	/// \sa glDrawElements()
+	/// \sa http://www.opengl-tutorial.org/intermediate-tutorials/tutorial-9-vbo-indexing/
+	/// \sa https://www.opengl.org/sdk/docs/man/html/glDrawElements.xhtml
 	virtual void drawElements(const ofVbo & vbo, GLuint drawMode, int amt, int offsetelements) const=0;
```

Comment:
I had significant trouble understanding how to describe the differences between regular drawing, drawing elements, drawing instanced, and drawing elements instanced, so for these purposes I erred on the side of being vague about their uses so as not to provide misinformation. Where appropriate I linked to the openGL references to each function. Perhaps folks that are more well-versed in these forms of drawing can provide better documentation for these functions.